### PR TITLE
code optimization, extract common code hooks usePrevious

### PR DESCRIPTION
**Pull request by nuoyz**
**Description**

code optimization: usePrevious code implementations exist in multiple files; Easy maintenance, common code implementation

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #???

### CODE REVIEW

**Comment by nuoyz** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by gantunesr — CHANGES_REQUESTED** [review]
Good catch! Can you refactor the hook code to TypeScript?

**Comment by nuoyz** [comment]
ok

File: app/components/hooks/usePrevious.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+export default function usePrevious(state) {
+	const ref = useRef();
+
+	useEffect(() => {
+		ref.current = state;
+	});
+
+	return ref.current;
+}
```
Comment (gantunesr) [inline app/components/hooks/usePrevious.js:11]: The JS file should be deleted

File: app/components/UI/Notification/index.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import SimpleNotification from './SimpleNotification';
 import { currentNotificationSelector } from '../../../reducers/notification';
 
 import { findRouteNameFromNavigatorState } from '../../../util/general';
+import usePrevious from "../../hooks/usePrevious";
```
Comment (gantunesr) [inline app/components/UI/Notification/index.js:13]:
```suggestion
import usePrevious from '../../hooks/usePrevious';
```

**Review by gantunesr — CHANGES_REQUESTED** [review]

**Comment by nuoyz** [comment]
what do i need to do next?

**Comment by rickycodes** [comment]
@nuoyz I've taken the liberty of squashing your commits (and properly attributing them). you're PR should be good to merge now :)

**Review by gantunesr — APPROVED** [review]

**Pull request merged**